homeassistant/components/squeezebox/browse_media.py

```diff
@@ -262,21 +263,30 @@ def _get_item_thumbnail(
 
     if track_id:
         if internal_request:
-            item_thumbnail = player.generate_image_url_from_track_id(track_id)
-        elif item_type is not None:
-            item_thumbnail = entity.get_browse_image_url(
-                item_type, item["id"], track_id
-            )
+            return cast(str, player.generate_image_url_from_track_id(track_id))
+        if item_type is not None:
+            return entity.get_browse_image_url(item_type, item["id"], track_id)
+
+    url = None
+    content_type = item_type or "unknown"
+
+    if search_type in ["apps", "radios"]:
+        url = cast(str, player.generate_image_url(item["icon"]))
```

Comment:
Same, the method is typed as `def generate_image_url(self, image_url: str) -> str:`